src/main/java/org/mockbukkit/mockbukkit/block/BlockMock.java

```diff
@@ -311,9 +311,7 @@ public BlockFace getFace(@NotNull Block block)
 	@Override
 	public @NotNull BlockState getState()
 	{
-		// This will always return a snapshot of the BlockState, not the actual state.
-		// This is optional with Paper but for Spigot it simply works like that.
-		return state.getSnapshot();
```

Comment:
This commit is kinda true. The Block state is always a new instance, but the inventory data is stored in the minecraft block which is always the same. So to emulate the inventory we need this to be constant, otherwise there's no other way to store the current state, since we don't work directly with minecraft entities.